gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/step/sideEffect/AggregateGlobalStep.java

```diff
@@ -122,8 +121,16 @@ protected Traverser.Admin<S> processNextStart() {
 
     @Override
     public void processAllStarts() {
+        final TraversalSideEffects sideEffects = this.getTraversal().getSideEffects();
+
+        // Pre-defined Operator such as addAll and assign will reduce over the whole input set, rather than
+        // applying a single input one by one.
```

Comment:
Suggest to also add to comment here on how both these operators work to be clear on why we focus on reduction over the whole input set. Basically, `addAll` can only work when either inputs to it are Maps or instances of Collection interface. And `assign` works for any type of input.

Reply:
Changed the comment, please take a look